plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -547,6 +550,30 @@ func (r *ReferenceManager) Validate(ctx context.Context, a admission.Attributes,
 									}
 								}
 							}
+							for _, imageVersion := range machineImage.Versions {
+								for _, imageFlavor := range imageVersion.CapabilityFlavors {
+									for _, removedCapability := range removedCapabilities {
+										if capabilityValues, exist := imageFlavor.Capabilities[removedCapability.Name]; exist {
+											for _, capabilityValue := range capabilityValues {
+												if slices.Contains(removedCapability.Values, capabilityValue) {
+													channel <- fmt.Errorf("unable to delete MachineCapability %q with value %q from CloudProfile %q - capability value is still in use by NamespacedCloudProfile '%s/%s'", removedCapability.Name, capabilityValue, cloudProfile.Name, nscpfl.Namespace, nscpfl.Name)
+												}
+											}
+										}
+									}
+								}
+							}
+						}
```

Comment:
Could you extract this section to eliminate the deep nesting in this file? Same for the next section.